ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivateStateRootResolver.java

```diff
@@ -33,6 +34,26 @@ public PrivateStateRootResolver(final PrivateStateStorage privateStateStorage) {
     this.privateStateStorage = privateStateStorage;
   }
 
+  public Hash resolveLastStateRoot(
+      final Bytes32 privacyGroupId, final PrivateMetadataUpdater privateMetadataUpdater) {
+    final PrivateBlockMetadata privateBlockMetadata =
+        privateMetadataUpdater.getPrivateBlockMetadata(privacyGroupId);
+    if (privateBlockMetadata != null) {
+      return privateBlockMetadata.getLatestStateRoot().get();
+    } else {
+      final Hash blockHashForLastBlockWithTx =
+          privateMetadataUpdater.getPrivacyGroupHeadBlockMap().get(privacyGroupId);
+      if (blockHashForLastBlockWithTx != null) {
+        return privateStateStorage
+            .getPrivateBlockMetadata(blockHashForLastBlockWithTx, privacyGroupId)
+            .flatMap(PrivateBlockMetadata::getLatestStateRoot)
+            .orElse(EMPTY_ROOT_HASH);
```

Comment:
Should this ever happen? If we have a hash on our `PrivacyGroupHeadBlockMap`, we should always find the state root for it, should we?

Maybe this should throw an error.

Reply:
I think you are right. I think I have copied that from the original code and did not really question that. I will fix both!

Reply:
You are right. Copied that from the original method without questioning it. Will fix both!